packages/components/src/components/Dropdown/Menu.tsx

```diff
@@ -0,0 +1,239 @@
+    position: relative;
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    gap: ${spacingsPx.sm};
+    padding: ${spacingsPx.xs} ${spacingsPx.sm};
+    border-radius: ${borders.radii.xs};
+    color: ${({ isDisabled, theme }) => (!isDisabled ? theme.textDefault : theme.textDisabled)};
+    white-space: nowrap;
+    transition: background 0.2s ease;
+    pointer-events: ${({ isDisabled }) => (isDisabled ? 'none' : 'auto')};
+    cursor: ${({ noHoverEffect }) => (noHoverEffect ? 'default' : 'pointer')};
+
+    > span {
+        margin-right: auto;
+    }
+
+    :hover {
+        background: ${({ theme, noHoverEffect }) =>
+            !noHoverEffect && theme.backgroundSurfaceElevation0};
+    }
+
+    ${({ separatorBefore, theme }) =>
+        separatorBefore &&
+        css`
+            margin-top: ${spacingsPx.md};
+
+            :after {
+                position: absolute;
+                width: 100%;
+                top: -${spacingsPx.xs};
+                left: 0;
+                border-top: 1px solid ${theme.borderOnElevation1};
+                content: '';
+            }
+        `}
+`;
+
+interface MasterLinkProps {
+    label: React.ReactNode;
+    icon: IconProps['icon'];
+    setToggled: (toggled: boolean) => void;
+    callback?: () => void;
+}
+
+const MasterLink = ({ label, icon, callback, setToggled }: MasterLinkProps) => {
+    const theme = useTheme();
+
+    return (
+        <MasterLinkComponent
+            onClick={() => {
+                if (callback) {
+                    callback();
+                    setToggled(false);
+                }
+            }}
+        >
+            <span>{label}</span>
+            <Icon icon={icon} size={spacings.sm} color={theme.iconPrimaryDefault} />
+        </MasterLinkComponent>
+    );
+};
+
+interface DropdownMenuItemProps {
+    key: string;
+    label: React.ReactNode;
+    callback?: () => any | Promise<any>;
+    icon?: IconProps['icon'];
+    iconRight?: IconProps['icon'];
+    isDisabled?: boolean;
+    isHidden?: boolean;
+    noHoverEffect?: boolean;
+    separatorBefore?: boolean;
+    'data-test'?: string;
+}
+
+const MenuItem = ({
+    icon,
+    iconRight,
+    label,
+    isDisabled,
+    callback,
+    setToggled,
+    ...rest
+}: DropdownMenuItemProps & { setToggled: (toggled: boolean) => void }) => {
+    const onMenuItemClick = (e: React.MouseEvent) => {
+        e.stopPropagation();
+
+        // Close the menu if item is not disabled and if
+        // a) callback func is not defined
+        // b) callback is defined and returns true/void
+        if (!isDisabled) {
+            if (callback) {
+                const shouldCloseMenu = callback();
+                if (shouldCloseMenu === true || shouldCloseMenu === undefined) {
+                    setToggled(false);
+                }
+            } else {
+                setToggled(false);
+            }
+        }
+    };
```

Comment:
I think this is a bit wild...a better API would be IMO: replacing `callback` with an `onClick` prop, removing `noHoverEffect`. If `onClick` is provided, call it here set `cursor: pointer` and close the menu. If `onClick` is defined on the item component, all of this can be handled by the item component. Up to you if you want to do it, I know it is old code...

If you do it this way, `callback` prop on `MasterLink` should also be renamed to `onClick`. Perhaps also rename `MasterLink` to `Addon` to make it consistent with `CollapsibleBox`.

Reply:
Good suggestions! I totally ignored that part, now i can't understand how :DD  
e623df335e6da737687211b693978dd154359f5e
073a743cc0